cloud_automation/java/dataflow/src/main/java/org/datacommons/dataflow/CsvImport.java

```diff
@@ -0,0 +1,121 @@
+    @ProcessElement
+    public void processElement(ProcessContext c) throws Exception {
+      try {
+        // TODO: Pass header as congfiguration.
+        String[] headers = new String[] {"key", "value"};
+        String[] values = c.element().split(",");
+        Preconditions.checkArgument(headers.length == values.length);
+
+        byte[] rowkey = Bytes.toBytes(values[0]);
+        byte[][] headerBytes = new byte[headers.length][];
+        ImmutableList.Builder<Mutation> mutations = ImmutableList.builder();
+        for (int i = 0; i < headers.length; i++) {
+          Mutation.SetCell setCell = null;
+          headerBytes[i] = Bytes.toBytes(headers[i]);
+          setCell =  Mutation.SetCell.newBuilder()
+                          .setFamilyName(FAMILY)
+                          .setColumnQualifier(ByteString.copyFrom( headerBytes[i]))
+                          .setValue(ByteString.copyFrom(Bytes.toBytes(values[i])))
+                          .build();
+           mutations.add(Mutation.newBuilder().setSetCell(setCell).build());
+        }
+        Iterable<Mutation> mutIter = mutations.build();
+        c.output(KV.of(ByteString.copyFrom(rowkey), mutIter));
+      } catch (Exception e) {
+        LOG.error("Failed to process input {}", c.element(), e);
+          throw e;
+      }
+    }
+  };
+
+  public static interface BigtableCsvOptions extends CloudBigtableOptions {
+
+    @Description("The headers for the CSV file.")
+    ValueProvider<String> getHeaders();
+
+    @SuppressWarnings("unused")
+    void setHeaders(ValueProvider<String> headers);
+
+    @Description("The Cloud Storage path to the CSV file.")
+    ValueProvider<String> getInputFile();
+
+    @SuppressWarnings("unused")
+    void setInputFile(ValueProvider<String> location);
+  }
+
+
+  /**
+   * <p>Creates a dataflow templae with following stages: </p>
+   * <ol>
+   * <li> Read a CSV from GCS.
+   * <li> Transform each line to a Bigtable Row mutation.
+   * <li> Write to Bigtable.
+   * </ol>
+   *
+   * @param args Arguments to use to configure the Dataflow Pipeline. Check README for deployment
+   * instructions.
+   * --project=[dataflow project] \\ --stagingLocation=gs://[your google storage bucket] \\
+   * --inputFile=gs://[your google storage object] \\
+   * --bigtableProject=[bigtable project] \\ --bigtableInstanceId=[bigtable instance id] \\
+   * --bigtableTableId=[bigtable tableName]
+   *
+   *
+   *  This pipepline uses BigtableIO instead of CloudBigtableIO libraries. This is because,
+   *  CloudBigtableIO libraries do not completely support dataflow templates.
+   *  The inspiration for this is pipeline is from cloud google ataflow examples.
+   */
+
+  public static void main(String[] args) throws IllegalArgumentException {
+    BigtableCsvOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigtableCsvOptions.class);
+
+    Pipeline p = Pipeline.create(options);
+
```

Comment:
I'm comparing to the original CsvImport.java. should we retain the options checks?

    if (options.getInputFile().equals("")) {
      throw new IllegalArgumentException("Please provide value for inputFile.");
    }
    if (options.getHeaders().equals("")) {
      throw new IllegalArgumentException("Please provide value for headers.");
    }

Reply:
Can I do this in a follow-up. 
Doing this gives warnings in the java code, there is something about the way templates get parameters that causes the warnings. I want to be sure I understand the warning before doing this. This is why I removed this.